File: OP_laba6_sharp/OP_laba6_sharp/Expression.cs
using System;

namespace OP_laba6_sharp
{
    public class Expression
    {
        private double A { get; set; }
        private double C { get; set; }
        private double D { get; set; }

        public Expression(double a, double c, double d)
        {    
            A = a;
            C = c;
            D = d;
        }

        public double Calculate()
        {
            
            if (A < 0 )
            {
                throw new ArithmeticException("Incorrect square root argument.");
            } 
            if (A == 4) 
            {
                throw new DivideByZeroException("Denominator shouldn`t be equal 0.");
            }
            double result = (2 * C - D + Math.Sqrt(23 * A)) / (A / 4 - 1); 
            return result;
        }

    }
}

File: OP_laba6_c++/src/Expression.cpp
#include "Expression.h"
#include <math.h>

Expression::Expression(double a, double c, double d) {
    this->_a = a;
    this->_c = c;
    this->_d = d;
}

double Expression::Calculate() {
    if (this->_a == 4) {
        throw "Divide by zero exception. Denominator shouldn`t be equal 0.";
    }
    if (this->_a < 0) {
        throw "Argument Exception. Square root argument should be more than 0";
    }
    double result = (2 * this->_c - this->_d + sqrt(23*this->_a)) / (this->_a/4 - 1);
    return result;
}
